presto-hive/src/test/java/com/facebook/presto/hive/TestSelectiveHiveDistributedQueries.java

```diff
@@ -0,0 +1,100 @@
+package com.facebook.presto.hive;
+
+import com.facebook.presto.tests.AbstractTestDistributedQueries;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.hive.HiveQueryRunner.createQueryRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.tpch.TpchTable.getTables;
+
+public class TestSelectiveHiveDistributedQueries extends AbstractTestDistributedQueries {
+
+    protected TestSelectiveHiveDistributedQueries() {
+        super(() -> createQueryRunner(getTables(), ImmutableMap.of(), getHiveProperties(), Optional.empty()));
+    }
+
+    private static Map<String, String> getHiveProperties() {
+        HashMap<String, String> properties = new HashMap<>();
+        properties.put("hive.pushdown-filter-enabled", "true");
+        return properties;
+    }
+
+    @Override
+    protected boolean supportsNotNullColumns()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsRowByRowDelete()
+    {
+        return false;
+    }
+
+    @Override
+    @Test
+    public void testExistsSubquery() {
+        // cannot pushdown filter that is always false
+    }
+
+    @Override
+    @Test
+    public void testGroupByKeyPredicatePushdown() {
+        // partial filters are not supported
+    }
+
+    @Override
+    @Test
+    public void testScalarSubquery() {
+        // cannot pushdown filter that is always false
+    }
+
+    @Override
+    @Test
+    public void testSemiJoin() {
+        // partial filters are not supported
+    }
+
+    @Override
+    @Test
+    public void testSemiJoinWithGroupBy() {
+        //  partial filters are not supported
+    }
+
```

Comment:
Why are you overriding these in this way, rather than extending the ```supportsXYZ() ``` pattern that seems to be emerging?

Does this pull request simply serve as a sort of documentation that keeps track of what we do and don't have able to run, yet? What does this commit accomplish?

Reply:
I didn't extend the pattern here since I was less certain whether the reason for omitting these tests was correct, but that probably isn't good enough a reason.  I should probably be consistent.

What this pull request accomplishes is that it adds test coverage for pushdown-filter-enabled.  We are still applying some 600 more tests than with a few thousand queries.  It also documents a bug.

Reply:
The other reasons I don't extend the pattern is because these tests come from AbstractTestQueries.java and not from AbstractTestDistributedQueries.java and the "cannot pushdown filter that is always false" exception doesn't make any sense to me.  Maybe it is a sensible exception, but I don't want to enforce something that I don't understand.